### ISSUE TITLE
npm help <term> always showing "Top hits" list

### ISSUE THREAD

**Issue by yanxyz** [npm/npm#7689]
After I install io.js 1.6.1 which contains npm 2.7.1, npm help <term> always showing "Top hits" list. Then I delete the file npm and npm.cmd in the io.js dir and install the older version npm.  The problem remains under npm@2.7.0. The problem is gone under npm@2.6. My OS is win8.1 x64.

**Comment by KenanY** [comment]
@yanxyz I can't reproduce on Linux, but this might be system specific. Probably a result of https://github.com/npm/npm/pull/4595, which was introduced in `npm@2.7.0`. Could you test each of the following commits to see if I'm correct on this?

https://github.com/npm/npm/compare/628fcdb0be4e14c0312085a50dc2ae01dc713fa6...d3cb6e1ff82d8cae7610b6c0ad751913d09846b6

  - https://github.com/npm/npm/commit/84b7b809b1cc74912be85247c4becf289a8e0c16 is the commit just before the merge
  - https://github.com/npm/npm/commit/50f461d248c4d22e881a9535dccc1d57d994dbc7 is the change in question
  - https://github.com/npm/npm/commit/d3cb6e1ff82d8cae7610b6c0ad751913d09846b6 is an update to the change which might be relevant here

You can install each commit with `npm install npm/npm#commit`.

**Comment by yanxyz** [comment]
All have the same result, showing the "Top hits" list.

![image](https://cloud.githubusercontent.com/assets/737120/6751822/a73b5fd8-cf43-11e4-94e5-260716ebe4cf.png)

**Comment by yanxyz** [comment]
I test io.js 1.6.1/npm 1.7.1 in other systems. It's ok  under Ubuntu, it has the problem under xp.

**Comment by yanxyz** [comment]
well, I find out this problem is caused by these lines  in `help.js`:
```js
var compext = "\\.+(gz|bz2|lzma|[FYzZ]|xz)$"
var f = "+(npm-" + section + "|" + section + ").[0-9]?(" + compext + ")"
```
And it is fixed by 1ab57c38116c0403965c92bf60121f0f251433e4. Just wait for npm 2.10.0.
